### ISSUE TITLE
Request: way for amp-instagram embeds to resize themselves

### ISSUE THREAD

**Issue by rafaelhbarros** [ampproject/amphtml#5032]
Please walk through the following steps for reporting an issue about AMP:

Instagram embeds that contain height bigger than the default will overflow with a scrollbar.
It would be nice if it could resize itself after loading.

Unfortunately, we don't store the dimensions of the image on our CMS ahead of time, and loading them on a per-request basis is too expensive.
## How do we reproduce the issue?

Here is an example of embed that causes the effect:

```
<amp-instagram width="400" height="400" data-shortcode="/p/BJ5sKT0g4S7/" layout="responsive">
```

**Comment by poscar** [comment]
@jasti Here's the issue mentioned on our call.  Let us know if you have any questions or need any other details.

Thanks!

**Comment by cramforce** [comment]
To fix this we'd need a contact with instagram.
CC @adewale in case he knows anyone there.

**Comment by adewale** [comment]
I've let Instagram know.

**Comment by poscar** [comment]
Any updates on this one? :)

**Comment by adewale** [comment]
Sorry but we haven't made any progress with the Instagram folks. I'll ping them again.

**Comment by src-code** [comment]
Has this one hit a dead-end?  I'd love to have this one resolved, as we too have large amounts of Instagram-embedded content without image dimension data, so we're currently forced to assume a worst-case portrait aspect ratio, or suffer scroll bars.

**Comment by cramforce** [comment]
I took a look and instagram is sending the data already (Test case http://output.jsbin.com/tozokoj/quiet See console).

So, this would be a simple change to amp-instagram.js

**Comment by src-code** [comment]
@cramforce Would it just be a matter of doing something like what `amp-list` does, taking a measurement of the content and requesting a height adjustment after the iframe load promise is resolved?  eg:

```js
      // Change height if needed.
      this.getVsync().measure(() => {
        const scrollHeight = this.container_./*OK*/scrollHeight;
        const height = this.element./*OK*/offsetHeight;
        if (scrollHeight > height) {
          this.attemptChangeHeight(scrollHeight).catch(() => {});
        }
      });
```

In that case, it doesn't seem we'd actually need the measurements from Instagram?

**Comment by src-code** [comment]
Am I right in thinking this component would have to be wrapped in a 3p iframe ala Twitter/Facebook in order to properly map the messages to the specific embed instance?  I might make an attempt at implementing this one...

**Comment by cramforce** [comment]
@src-code Nope, no need for major changes. Just install a postMessage receiver from the current component (similar to amp-youtube and other video player) and handle the message they are sending.

**Comment by src-code** [comment]
Ok, took a stab at implementing this one (#7705), would love feedback!

**Comment by jridgewell** [comment]
Closed by #7705.
